doc/languages-frameworks/beam.section.md

```diff
@@ -56,11 +56,117 @@ Mix functions similarly to Rebar3, except we use `buildMix` instead of `buildReb
+
+Here is how your default.nix file would look like
+
+```default.nix
+with import <nixpkgs-unstable> { };
+
+let
+  packages = beam.packagesWith beam.interpreters.erlang;
+  src = builtins.fetchGit {
+    url = "ssh://git@github.com/your_id/your_repo";
+    rev = "replace_with_your_commit";
+  };
+
+  nodeDependencies =
+    (pkgs.callPackage ./assets/default.nix { }).shell.nodeDependencies;
+
+in packages.buildMix {
+  inherit src;
+  name = "your_project";
+  version = "0.0.1";
+  mixEnv = "prod";
+  # leave this empty and nix will complain and tell you the right value
+  # to replace this with
+  depsSha256 = "";
+  inherit src;
+  # if you have build time environment variable add them here
+  buildEnvVars = {
+    MY_ENV_VAR="my_value";
+  };
+  preConfigure = ''
+    cd ./assets
+
+    ln -s ${nodeDependencies}/lib/node_modules ./node_modules
+    export PATH="${nodeDependencies}/bin:$PATH"
+
+    webpack --mode production --config ./webpack.config.js
+    cd ..
+  '';
+}
+```
+
+Setup will require the following steps
+
+- Keep in mind that having secrets in the build time environment variables is not recommended. If those are not secrets they can be directly included in the config.
+- move your secrets to runtime envionment variables. For more information about this check [runtime.exs docs](https://hexdocs.pm/mix/Mix.Tasks.Release.html#module-runtime-configuration). On a fresh Phoenix build that would mean that both of the `DATABASE_URL` and `SECRET_KEY` need be moved to `runtime.exs`.
+- `cd assets` and `node2nix --development` will generate a nix expression containing your frontend dependencies
+- commit and push those changes
+- you can now `nix-build .`
+- When you need to run the release, you will need to at least supply the environment variable RELEASE_TMP that you have write access to.
+
+#### Example of creating a service for an Elixir - Phoenix project
+
+In order to create a service with your release, you could add a `service.nix`
+in your project with the following
+
+```Nix
+{config, pkgs, lib, ...}:
+
+let
+  release = pkgs.callPackage ./default.nix;
+  release_name = "app";
+  working_directory = "/home/app";
+in
+{
+  systemd.services.${release_name} = {
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network.target" "postgresql.service" ];
+    requires = [ "network-online.target" "postgresql.service" ];
+    description = "my app";
+    serviceConfig = {
+      Type = "exec";
+      DynamicUser = true;
+      WorkingDirectory = working_directory;
+      # Implied by DynamicUser, but just to emphasize due to RELEASE_TMP
+      PrivateTmp = true;
+      ExecStart = ''
+        ${release}/bin/${release_name} start
+      '';
+      ExecStop = ''
+        ${release}/bin/${release_name} stop
+      '';
+      ExecReload = ''
+        ${release}/bin/${release_name} restart
+      '';
+      Restart = "on-failure";
+      RestartSec = 5;
+      StartLimitBurst = 3;
+      StartLimitInterval = 10;
+      environment = {
```

Comment:
environment shouldn't be in serviceConfig

it needs to be directly in the service, like so: `systemd.services.${release_name}.environment = {}`

Reply:
Thanks!